pkg/foundation/cerrors/fatal.go

```diff
@@ -0,0 +1,41 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cerrors
+
+// FatalError is an error type that will diferentiate these from other errors that could be retried.
+type FatalError struct {
+	Err error
+}
+
+// NewFatalError creates a new FatalError.
+func NewFatalError(err error) *FatalError {
```

Comment:
I believe the go convention for errors is just having `FatalError` instead of `NewFatalError`, I also noticed the main issue has it as `FatalError` too, maybe think of a way we could accomplish this? maybe renaming the struct to `FatalErr` instead? not sure of the right way to do so but it's worth the thought

Reply:
I started defining `FatalError` as the constructor, but I wasn't in love with any name I gave to the struct type. I also noticed we _break_ that convention in places such as:

- https://github.com/ConduitIO/conduit/blob/06bd6e58cb1ff17841319897109dfade369a5ddd/pkg/plugin/connector/mock/plugin.go#L33
- https://github.com/ConduitIO/conduit/blob/76c25f1d1c140e65164e5ec5147a0fbe5d7e2830/pkg/plugin/connector/service.go#L62
- https://github.com/ConduitIO/conduit/blob/817d52c3184f33e7b4b542afc8b794196a68c420/pkg/conduit/runtime.go#L109

And a long etc.

I'll make the change however to meet the criteria of the issue, as you mentioned, but I'd be curious to see what's the real convention around this.

Reply:
Looks like the type actually needs to end as `Error`: https://github.com/ConduitIO/conduit/actions/runs/10613375269/job/29417037570?pr=1811#step:5:25.

I think I'll change it back the way it was before.

Reply:
@maha-hajja in a second thought, I made [this modification](https://github.com/ConduitIO/conduit/pull/1811/commits/44652268b6a635c1a20a965ab61977c6ac1bf075). Let me know what you think.

Reply:
Interesting. It seems like this is still required:

```
pkg/foundation/cerrors/fatal.go:22:6: the type name `Fatal` should conform to the `XxxError` format (errname)
```

Reply:
thank you for trying! ok let's keep it as `NewFatalError`, since there's no pretty way to do that now